### DISCUSSION TITLE
DB Restore Automation

### DISCUSSION THREAD

**Discussion by OzzyMtl** [dataplat/dbatools#7094] [General]
Hello,

We have just found out about DBATools and would like to use it for automated Restores.
I couldn't find examples or switches matching with our scenario. Can somebody help please? Thanks in advance.

Ozzy

Script should Get DB full backup file \\server1\fullbackups\DB_name\
and also Get DB differential backup file \\server2\diff\DB_Name\
then Restore on one of the SQL Server on the network with Replace (overwrite)

There are multiple files under these folders and script should get only the latest (the newest) file available.

**Comment by wsmelton** [comment]
Reference: https://docs.dbatools.io/#Restore-DbaDatabase

We have multiple examples of pointing the restore command at a path for it to scan. It supports the use of Ola Hallengren's backup directory structure. If you have a backup folder structure where database files are spread across different file servers then we cannot support that and you will need to gather the file objects and provide what we should scan (_see Example 9_).
